src/StatsdClient/StatsdConfig.cs

```diff
@@ -67,6 +88,10 @@ public StatsdConfig()
             StatsdPort = 0;
             StatsdMaxUDPPacketSize = DefaultStatsdMaxUDPPacketSize;
             Advanced = new AdvancedStatsConfig();
+
+            Service = System.Environment.GetEnvironmentVariable(ServiceEnvVar);
```

Comment:
I'm not sure how defensive we need to be about `GetEnvironmentVariable` calls throwing a `SecurityException` since I see many other of these calls in the codebase that are not wrapped in try/catch blocks, but that came to mind when I read this block